src/components/Button/Button.tsx

```diff
@@ -74,14 +93,55 @@ class Button extends UIComponent<IButtonProps, any> {
     classes,
     rest,
   }: IRenderResultConfig<IButtonProps>): ReactNode {
-    const { children, content } = this.props
+    const { children, content, disabled, icon, iconPosition, type } = this.props
+    const primary = type === 'primary'
+
+    const getContent = (): ReactNode => {
+      if (childrenExist(children)) {
+        return children
+      }
+
+      const iconIsAfterButton = iconPosition === 'after'
+      const renderedContent = [
+        content && <Text key="btn-content" truncated content={content} />,
+        icon &&
+          typeof icon === 'string' && (
+            <Icon
+              key="btn-icon"
+              name={icon}
+              xSpacing={!content ? 'none' : iconIsAfterButton ? 'before' : 'after'}
+              color={primary ? 'white' : 'black'}
```

Comment:
If we want the icon color to be the same as the text color, we can use 'currentColor' instead of the hard-coded value.

Reply:
@notandrew 
`currentColor` would be a hex number right now and we're not supporting that for `Icon` component yet. I think @kuzhelov has an ongoing PR for that; we'll address this afterwards
fyi @mnajdova